sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -221,4 +221,40 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
       emptyTableData.agg(sumDistinct('a)),
       Row(null))
   }
+
+  test("moments") {
+    checkAnswer(
+      testData2.agg(skewness('a)),
+      Row(0.0))
+
+    checkAnswer(
```

Comment:
I think sharing `checkAnswerWithTol` as a helper function in `QueryTest` sounds like a good plan, I'm not super sure on the name (since checkAnswer works with an expected Row and the signature is a bit further off than just adding a tolerance param), but it seems probably ok with the current name.